### PR TITLE
KEYCLOAK-18073 avoid ModelDuplicateException during parallel starup o…

### PR DESCRIPTION
**Pull request by vramik**
…f servers

### CODE REVIEW

File: model/map/src/main/java/org/keycloak/models/map/dblock/NoLockingDBLockProviderFactory.java [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+import org.keycloak.common.Profile;
+import org.keycloak.models.KeycloakSession;
+import org.keycloak.models.KeycloakSessionFactory;
+import org.keycloak.models.dblock.DBLockProvider;
+import org.keycloak.models.dblock.DBLockProviderFactory;
+import org.keycloak.provider.EnvironmentDependentProviderFactory;
+
+public class NoLockingDBLockProviderFactory implements DBLockProviderFactory, EnvironmentDependentProviderFactory { 
+
+    public static final String PROVIDER_ID = "map";
```
Comment (hmlnarik) [inline model/map/src/main/java/org/keycloak/models/map/dblock/NoLockingDBLockProviderFactory.java:30]:
```suggestion
    public static final String PROVIDER_ID = "none";
```

File: testsuite/integration-arquillian/tests/base/pom.xml [hunk 2]
```diff
@@ -1188,6 +1188,7 @@
                             <systemPropertyVariables>
                                 <keycloak.profile.feature.map_storage>enabled</keycloak.profile.feature.map_storage>
                                 <keycloak.mapStorage.provider>concurrenthashmap</keycloak.mapStorage.provider>
+                                <keycloak.dblock.provider>map</keycloak.dblock.provider>
```
Comment (hmlnarik) [inline testsuite/integration-arquillian/tests/base/pom.xml:1191]:
```suggestion
                                <keycloak.dblock.provider>none</keycloak.dblock.provider>
```

**Review by hmlnarik — CHANGES_REQUESTED** [review]
Thank you @vramik! Oveall looks good, the only requested change is in naming the db lock, as it has nothing to do with map. Have you considered placing this implementation into server-spi-private module?

**Comment by vramik** [comment]
https://master-jenkins/job/universal-test-pipeline-server/443

**Review by hmlnarik — APPROVED** [review]
Thank you for your contribution!

**Pull request merged**